pkg/cmd/image/tag.go

```diff
@@ -59,17 +60,23 @@ func Tag(ctx context.Context, client *containerd.Client, options types.ImageTagO
 	}
 	defer done(ctx)
 
-	image, err := imageService.Get(ctx, srcName)
+	img, err := imageService.Get(ctx, srcName)
 	if err != nil {
 		return err
 	}
-	image.Name = target.String()
-	if _, err = imageService.Create(ctx, image); err != nil {
+	img.Name = target.String()
+
+	err = imgutil.EnsureAllContent(ctx, client, srcName, img.Target, "", options.GOptions)
+	if err != nil {
+		return err
+	}
```

Comment:
When tagging, do we need to return errors for failed pulls? This might not meet expectations.

Reply:
Yes, that is a valid concern.
Then if at a later point we try to push it, it will fail…
Not sure how to deal with that…

Reply:
I do not like any of the code involved in there.
The "single-platform tmp-reduced-platform image" gymnastic / convert intermediary step on push just smells fishy.
Then again, this is containerd design issues here.

Reply:
We could introduce a new flag called `ensure-image` for both the `tag` and `commit` commands. 

When this flag is set, we would attempt to pull any missing layers. If any errors occur during this process, they would be returned.

Reply:
The thing that is bugging me with the new flag approach is that it would leave the default behavior broken, which I don't think is right.

So, what about this: instead of _hard error_ when the fetch fail, we just print a loud warning but we continue?
wdyt @lingdie ?

Reply:
Latest push implements this.
We no longer hard error, but warn the user.
PTAL at your convenience @lingdie

Reply:
look good to me.